controller/index.js

```diff
@@ -1,1019 +1,1054 @@
+    agol.find(req.params.id, function (err, data) {
       if (err) {
-        res.status(404).send( err );
-      } else {
-        // sort the req.query before we hash so we are consistent 
-        var sorted_query = {};
-        _(req.query).keys().sort().each(function (key) {
-          if (key != 'url_only' && key != 'format'){
-            sorted_query[key] = req.query[key];
-          }
-        });
-        // build the file key as an MD5 hash that's a join on the paams and look for the file 
-        var toHash = req.params.item + '_' + ( req.params.layer || 0 ) + JSON.stringify( sorted_query );
-        var key = crypto.createHash('md5').update(toHash).digest('hex');
-        // Get the item 
-
-        // set a really high limit so large datasets can be turned into feature services 
-        req.query.limit = req.query.limit || req.query.resultRecordCount || 1000000000;
-        req.query.offset = req.query.resultOffset || null;
-        agol.getItemData( data.host, req.params.id, req.params.item, key, req.query, function(error, itemJson){
-          if (error) {
-            if (error.code && error.error){
-              res.status(error.code).send( error.error );
-            } else {
-              res.status(500).send( error );
-            }
-          } else {
-            // pass to the shared logic for FeatureService routing
-            delete req.query.geometry;
-            delete req.query.where;
-            controller.processFeatureServer( req, res, err, itemJson.data, callback);
-          }
-        });
+        return res.status(404).send(err)
       }
-    });
-     
-  };
 
+      // sort the req.query before we hash so we are consistent
+      var sorted_query = {}
+      _(req.query).keys().sort().each(function (key) {
+        if (key !== 'url_only' && key !== 'format') {
+          sorted_query[key] = req.query[key]
+        }
+      })
+      // build the file key as an MD5 hash that's a join on the paams and look for the file
+      var toHash = req.params.item + '_' + (req.params.layer || 0) + JSON.stringify(sorted_query)
+      var key = crypto.createHash('md5').update(toHash).digest('hex')
+      // Get the item
+
+      // set a really high limit so large datasets can be turned into feature services
+      req.query.limit = req.query.limit || req.query.resultRecordCount || 1000000000
+      req.query.offset = req.query.resultOffset || null
+      agol.getItemData(data.host, req.params.id, req.params.item, key, req.query, function (error, itemJson) {
+        if (error) {
+          return res.status(error.code || 500).send(error.error || error)
+        }
+
+        // pass to the shared logic for FeatureService routing
+        delete req.query.geometry
+        delete req.query.where
+        controller.processFeatureServer(req, res, err, itemJson.data, callback)
+      })
+    })
+  }
+
+  /**
+   * Handles request for thubmnails
+   */
+  controller.thumbnail = function (req, res) {
```

Comment:
@dmfenton FYI i did not touch the tile or thumbnail methods as you are correct, its not used. I dont to blow them away though.